IBPSA/Fluid/HeatPumps/ModularReversible/BaseClasses/CalculateEfficiency.mo

```diff
@@ -0,0 +1,120 @@
+within IBPSA.Fluid.HeatPumps.ModularReversible.BaseClasses;
+model CalculateEfficiency
+  "Calculate the COP or EER of a device"
+  extends Modelica.Blocks.Icons.Block;
+  parameter Modelica.Units.SI.Power PEleMin(min=Modelica.Constants.eps)
+    "If eletrical power consumption falls below this value, COP will be set to zero";
+
+  Modelica.Blocks.Interfaces.RealInput PEle(final unit="W", displayUnit="W")
+    "Electrical power consumed by the system" annotation (Placement(
+        transformation(extent={{-140,-90},{-100,-50}}), iconTransformation(extent={{-140,
+            -90},{-100,-50}})));
+  Modelica.Blocks.Interfaces.RealInput QUse_flow(final unit="W", displayUnit="W")
+    "Useful heat flow" annotation (Placement(transformation(
+          extent={{-140,50},{-100,90}}), iconTransformation(extent={{-140,50},{-100,
+            90}})));
+  Modelica.Blocks.Interfaces.RealOutput COP(min=0, final unit="1")
+    "Output for calculated COP value"
+    annotation (Placement(transformation(extent={{100,50},{120,70}})));
+  Modelica.Blocks.Logical.Hysteresis hys(
+    uLow=PEleMin,
+    uHigh=PEleMin*1.1)
+    "Hysteresis to switch between calculation and no calculation"
+    annotation (Placement(transformation(extent={{-60,-50},{-40,-30}})));
+  Modelica.Blocks.Interfaces.BooleanInput hea
+    "=true for heating, false for cooling" annotation (Placement(transformation(
+          extent={{-140,-20},{-100,20}}), iconTransformation(extent={{-140,-20},{-100,
+            20}})));
+  Modelica.Blocks.Interfaces.RealOutput EER(min=0, final unit="1")
+    "Output for calculated EER value"
+    annotation (Placement(transformation(extent={{100,-70},{120,-50}})));
+protected
+  Modelica.Blocks.Math.Division copCom "Computes COP"
+    annotation (Placement(transformation(extent={{10,40},{30,60}})));
+  Modelica.Blocks.Logical.Switch swi
+    "Outputs COP or zero depending on electricity use"
+    annotation (Placement(transformation(extent={{-4,-50},{16,-30}})));
+  Modelica.Blocks.Sources.Constant zer(final k=0) "Outputs zero"
+    annotation (Placement(transformation(extent={{-40,-80},{-20,-60}})));
+  Modelica.Blocks.Logical.Switch swiDivZer
+    "Outputs COP or zero depending on electricity use"
```

Comment:
No, was not aware of this function. I used this instead, but still keeping the switch to zero. Or do you mean to actually calculate the COP / EER near zero?

Reply:
I was thinking also computing the COP near zero power with `InverseXRegularized` but I can see how this could yield some unreasonably high COP values. This would need further regularization and testing, probably not worth it.